framework/web/Request.php

```diff
@@ -913,15 +989,33 @@ public function getReferrer()
      */
     public function getUserAgent()
     {
-        return isset($_SERVER['HTTP_USER_AGENT']) ? $_SERVER['HTTP_USER_AGENT'] : null;
+        return $this->headers->get('User-Agent');
     }
 
     /**
      * Returns the user IP address.
+     * The IP is determined using headers and / or `$_SERVER` variables.
+     * @since 2.0.13
      * @return string|null user IP address, null if not available
      */
     public function getUserIP()
     {
+        foreach ($this->ipHeaders as $ipHeader) {
+            if ($this->headers->has($ipHeader)) {
+                return trim(explode(',', $this->headers->get($ipHeader))[0]);
+            }
+        }
+        return $this->getRemoteIP();
```

Comment:
from above:

> Default is to trust all headers from all hosts.

you change the default value for userIP to be taken form a header by default, which was using the server remote address before, so installations that upgrade Yii will now become insecure by default.